### PR TITLE
Pull up to speed

### PR DESCRIPTION
**Pull request by pelson**
Closes https://github.com/conda-forge/artview-feedstock/pull/6

### CODE REVIEW

**Comment by pelson** [comment]
Interesting. Does artview support py3k? 

```
===== testing package: artview-1.2.3-py34_0 =====
import: 'artview'
/opt/conda/envs/_test/lib/python3.4/site-packages/pyart/io/grid_io.py:409: DeprecationWarning: read_legacy_grid is deprecated and will be removed in a future version of Py-ART
  "version of Py-ART", DeprecationWarning)
Traceback (most recent call last):
  File "/feedstock_root/build_artefacts/test-tmp_dir/run_test.py", line 25, in <module>
    import artview
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/artview/__init__.py", line 57, in <module>
    from . import scripts
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/artview/scripts/__init__.py", line 28, in <module>
    tmp = importlib.import_module('.'+module[:-3], __package__)
  File "/opt/conda/envs/_test/lib/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/artview/scripts/artview_old.py", line 295
    print "IN DISCONNECT"
                        ^
SyntaxError: Missing parentheses in call to 'print'
```

**Comment by jakirkham** [comment]
That's weird. Though this PR ( https://github.com/conda-forge/artview-feedstock/pull/5 ) passed. Hmm... In any event, merged that other re-rendering PR so we are using the `condaforge/linux-anvil` docker image.

**Comment by jakirkham** [comment]
Should this be closed?

**Comment by pelson** [comment]
> Should this be closed?

I don't think so - it improves the recipe, and also makes the feedstock upload correctly. I have rebased - let's see how it gets on.

**Comment by jakirkham** [comment]
Yeah, I think this needs to skip Python 3.x. I don't know how this has gone on for so long without being caught, but it is clearly Python 2.x only. We should probably remove any Python 3.x packages of this.

**Comment by pelson** [comment]
Ping @jjhelmus.

**Comment by jjhelmus** [comment]
Let me check with the package authors. @nguy @gamaanderson does ARTView support Python 3?

**Comment by jakirkham** [comment]
Their [code](https://github.com/nguy/artview/search?p=1&q=print&utf8=%E2%9C%93) looks much better. So, maybe we just need an upgrade or some patches.

**Comment by jakirkham** [comment]
Actually, this is super weird. The file it is failing on in CI `artview/scripts/artview_old.py` doesn't exist. Also, the line that is referenced doesn't exist. FWIW I'm on tag `Release_v1.2`, which is being used here. So, I have no idea how that is happening on CI. Does it rewrite its own code?!

**Comment by jakirkham** [comment]
Yeah, this file is definitely being generated somehow, but I'm really unclear as to how. Whatever is generating it is make it Python 2.x. AFAICT this is the only uncompliant file.

**Comment by jjhelmus** [comment]
Let me see if I can track it down.  I have some experience with the code base.

**Comment by jakirkham** [comment]
Sorry it just got more and more interesting the more I uncovered.

**Comment by jjhelmus** [comment]
I don't think the tar.gz in the GitHub release corresponds to files in the repo of the tagged release.

**Comment by jakirkham** [comment]
You're right. I suppose the less outlandish theory is usually correct. 😆

Sure enough `artview_old.py` does live in the `.tag.gz`. I suppose it can just be removed.

**Comment by jjhelmus** [comment]
The [Release_v1.2.tar.gz file](https://github.com/nguy/artview/archive/Release_v1.2.tar.gz) does not include artview_old.py and similar files.  I think that file should be used in the recipe.

**Comment by jakirkham** [comment]
I picked up this one ( https://github.com/nguy/artview/releases/download/Release_v1.2/artview-1.2.3.tar.gz ), which is what we are building with in this PR. It does have `artview_old.py`.

**Comment by jakirkham** [comment]
Is that not suppose to be an official release then?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 package:
   name: artview
-  version: "1.2"
+  version: {{ version }}
 
 source:
-  git_url: https://github.com/nguy/artview
-  git_rev: v1.2
+  fn: artview-{{ version }}.tar.gz
+  url: https://github.com/nguy/artview/releases/download/Release_v1.2/artview-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipe/meta.yaml:9]: Could we change this to `https://github.com/nguy/artview/archive/v{{ version }}.tar.gz`?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,13 +1,17 @@
+{% set version = "1.2.3" %}
```
Comment (jakirkham) [inline recipe/meta.yaml:1]: Also, this should be changed back to `1.2`.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
   name: artview
-  version: "1.2"
+  version: {{ version }}
 
 source:
-  git_url: https://github.com/nguy/artview
-  git_rev: v1.2
+  fn: artview-{{ version }}.tar.gz
+  url: https://github.com/nguy/artview/releases/download/Release_v1.2/artview-{{ version }}.tar.gz
+  sha256: 9fd8c78439f9a3867a1a4579c5430f9b0338d1b7652e34ccb3e7a008a202b4b2
```
Comment (jakirkham) [inline recipe/meta.yaml:10]: Not sure what this will need to be.

**Comment by jjhelmus** [comment]
I think it should be... but I think something got messed up.  I would use the tarbal of the repo.

@nguy can you clarify what file we should be using for the build?

File: recipe/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipe/meta.yaml:10]: Looks like it should be `db88e269a319b42781fab6016b029c3e07446cc7899135d78b46a62e4ae512ca`.

**Comment by jakirkham** [comment]
I went ahead and added this PR ( https://github.com/conda-forge/artview-feedstock/pull/6 ), which builds off this one, but switches back to `1.2` as it doesn't seem to have the same problem. This way you can get the tokens that @pelson nicely added while the rest of this gets sorted.

**Comment by gamaanderson** [comment]
artview should work in Python 3, and this `artview_old.py` should not exist.

**Comment by gamaanderson** [comment]
Version 1.2.3 is the official 1.2 release, we had some problems figuring out how to number the versions, that is why the releases are a little irregular.

That however does not explain why the `artview_old.py` file is in the download file, but not in the source code. Is it possible that the download files were created from the user directory and not from the git map tree? @nguy can you try to correct that?

In the mean while, try using the following file, this appears to be only the files mapped in git. 
https://github.com/nguy/artview/archive/Release_v1.2.tar.gz

**Comment by jakirkham** [comment]
Thanks for clarifying @gamaanderson. That's what PR ( https://github.com/conda-forge/artview-feedstock/pull/6 ) will do.

**Comment by nguy** [comment]
Yeah, I think this is my screw up. I  have files in the directory that are not tracked, but I probably made a tarball on the directory. I just cleaned those out, remade the tarball and updated the release. Hopefully that will help.

**Comment by jakirkham** [comment]
Restarted CIs.

**Comment by pelson** [comment]
@jakirkham - this PR is done with, right?

**Comment by jakirkham** [comment]
Maybe not. It sounds like the may have hot-fixed this tag so it should work. Also, it sounds like we should be using this tag if possible.

**Comment by jakirkham** [comment]
Sorry, @pelson, I should have closed mine and noted that we need to change the sha here as the tarball should be fixed. To save you time, I have gone ahead and done this in PR ( https://github.com/conda-forge/artview-feedstock/pull/7 ). Sorry again for the confusion.

**Pull request closed without merging**